src/lib/device-types/storage/gh.ts

```diff
@@ -0,0 +1,123 @@
+import * as Octokit from '@octokit/rest';
+
+import * as Bluebird from 'bluebird';
+import gunzip = require('gunzip-maybe');
+import * as _ from 'lodash';
+import * as request from 'request';
+import * as tar from 'tar-stream';
+import { captureException } from '../../../platform/errors';
+
+// import * as pkg from '../../../../package.json';
+
+import {
+	DEVICE_TYPE_REPO_NAME,
+	DEVICE_TYPE_REPO_OWNER,
+	DEVICE_TYPE_REPO_REF,
+} from '../../config';
+
+interface FetchDeviceTypeOptions {
+	owner: string;
+	repo: string;
+	archive_format?: string;
+	ref?: string;
+}
+
+interface DTContract {
+	slug: string;
+	version: string;
+	type: string;
+	name: string;
+}
+
+let cachedDeviceTypes: DTContract[] = [];
+let currentHash = '';
+
+export const fetchExternalDeviceTypes = async (
+	options: FetchDeviceTypeOptions = {
+		owner: DEVICE_TYPE_REPO_OWNER,
+		repo: DEVICE_TYPE_REPO_NAME,
+		ref: DEVICE_TYPE_REPO_REF,
+	},
+): Promise<string[]> => {
+	const octokit = new Octokit({
+		// ToDo: use current version string in UA
+		userAgent: `OpenBalenaApi`,
+	});
+	options.archive_format = 'tarball';
+	try {
+		const branch = await octokit.repos.getBranch({
+			owner: options.owner!,
+			repo: options.repo!,
+			branch: options.ref!,
+		});
+
+		if (branch.data.commit.sha !== currentHash) {
+			const archiveLinkResponse: AnyObject = await octokit.repos.getArchiveLink(
```

Comment:
Doesn't octokit have proper typings?
```suggestion
			const archiveLinkResponse = await octokit.repos.getArchiveLink(
```